subprojects/build-cache/src/main/java/org/gradle/caching/internal/origin/OriginReader.java

```diff
@@ -16,8 +16,9 @@
 
 package org.gradle.caching.internal.origin;
 
+import java.io.IOException;
 import java.io.InputStream;
 
 public interface OriginReader {
-    OriginMetadata execute(InputStream inputStream);
+    OriginMetadata execute(InputStream inputStream) throws IOException;
```

Comment:
would it make sense to make this `UncheckedIOException` instead of declaring it?

Reply:
I _like_ `IOException`. Well, in any case, I prefer using it naked to wrapping it in all kinds of things. This is a thing that reads something from an input stream, it has a right to throw an `IOException` without having to mess with wrapping things.